spicy/toolchain/src/compiler/codegen/parser-builder.cc

```diff
@@ -337,7 +337,7 @@ struct ProductionVisitor
                             builder()->addLocal("filtered_data", type::ValueReference(type::Stream()),
                                                 builder::id("filtered"));
                             args2[0] = builder::id("filtered_data");
-                            args2[1] = builder::default_(type::stream::Iterator()); // FIXME(bbannier): sensible data.
+                            args2[1] = builder::begin(builder::deref(args[0]));
```

Comment:
I'm wondering if this should be `args[2]` to be relative to the filtered stream? Both semantically, and also because otherwise we might end up comparing iterators for different streams?

Either way, a test using `offset()` on filtered data would be good too if we don't have one.

Reply:
> I'm wondering if this should be args[2] to be relative to the filtered stream?

Duh, good catch! Yes, this definitely was supposed to be `args2[0]` (unless I misunderstood your comment). 

> Either way, a test using offset() on filtered data would be good too if we don't have one.

We didn't have such a test, I added one. I do however have difficulty making it fail with my previous broken code. I discovered #1652 along the way though which might be a reason for that.